src/napari_workflows/_workflow.py

```diff
@@ -47,15 +47,7 @@ def set(self, name, func_or_data, *args, **kwargs):
         bound.apply_defaults()
         # Go through arguments and in case it's a callable, remove it
         # We should only have numbers, strings and images as parameters
-        used_args = [value for key, value in bound.arguments.items()]
-
-        for i in range(len(used_args)):
-            if callable(used_args[i]):
-                used_args[i] = None
-
-        # Remove None parameters from the end
-        while (used_args[-1] is None):
-            used_args = used_args[:-1]
+        used_args = [value for key, value in bound.arguments.items() if not callable(value)]
```

Comment:
Hi @multimeric ,

this is different from the proposed solution [here](https://github.com/haesleinhuepf/napari-workflows/issues/43#issuecomment-2290661585) and I'm afraid it breaks backwards compatbility. I would like to avoid this under all circumstances.

Have you tested this, e.g. with napari-assistant?

Thanks!

Best,
Robert

Reply:
Oh, but that functionality is the whole issue. If I can't prevent `None` from being removed from the end of the parameter list, it will never be possible to add `None` values to workflows.

Reply:
Currently, using the user-interface in the napari-assistant, it is not possible to add `None` as parameter. So where should these `None`s come from? In which context do they appear? Don't get me wrong, I'm just trying to solve issues which do not exist.

Reply:
Okay so the user will make an ordinary workflow that has no `None`. Then, "my" software will take this existing workflow and modify it programatically to add some extra steps and metadata. Some of this metadata might be "missing", but it still needs to be provided. However, adding a `None` value via `w.set("some_metadata", None)` causes `napari-workflows` to break, as noted.

Reply:
How about using dask graphs in your software and not napari-workflows?

Reply:
I think the issue is that using dask directly don't give us the GUI (`napari-assistant` etc) or the integrations with other napari plugins.